resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -542,7 +562,7 @@ public void start(BundleContext context) throws Exception {
 				workspaceInitCustomizer);
 		plugin = this; // must before open the tracker, as this can cause the registration of the
 						// workspace and this might trigger code that calls the static method then.
-		instanceLocationTracker.open();
+		new Thread(this::initialized, "Async ResourcesPlugin start").start(); //$NON-NLS-1$
```

Comment:
This is still wrong, the contract of the Resources plugin is that it tracks the workspace in the activator, as the activation usually (but that's not guaranteed!) happens by a lazy-activation and in this case must be guarded by the classloader lock.

Reply:
I agree with Cristoph. The code expected to have workspace initialized after activator is done, **not** asynchronously after that.

Please put breakpoint at `org.eclipse.core.internal.resources.Workspace.open(IProgressMonitor)` and check that this is executed in context of `org.eclipse.core.resources.ResourcesPlugin.start(BundleContext)` call.

Initialization of ResourcesPlugin should be changed very carefully, there are different actors playing here and not everything can be validated via automated tests.